### ISSUE TITLE
haddock-error gets buried under warnings

### ISSUE THREAD

**Issue by jwaldmann** [commercialhaskell/stack#2605]
I am using
```
stack build --haddock --haddock-deps --haddock-arguments=--hyperlinked-source
```
This fails, because haddock gives an error on one of the source files (ghc does not). But it is hard to find out from stack's output:
* last lines of stack output are just last lines of haddock warnings (and I have hundreds, like "missing documentation for ...") - I would expect a status message from stack saying that something failed
* so, to check the status, and see the error, I have to scroll back on the screen (or in the log file, whose location was printed by stack, but to see that, I also have to wade back throught the warnings)

**Comment by mgsloan** [comment]
Agreed that this should be addressed.  I think it will be easier to do so once we run haddock ourself, instead of delegating to Cabal.  That change is discussed in https://github.com/commercialhaskell/stack/issues/729
